src/borg/archive.py

```diff
@@ -225,19 +225,21 @@ def _preload(chunks):
                     # corresponding hardlink slave is selected (== is extracted).
                     # due to a side effect of the filter() call, we now have hardlink_masters dict populated.
                     for item in items:
-                        if 'chunks' in item:  # regular file, maybe a hardlink master
```

Comment:
this was the main issue, this only triggered for regular files (files which have content chunks).

a fifo file does not have content, but can be hardlinked to.